Remotion/ObjectBinding/Web.Development.WebTesting/ControlObjects/BocListValidationError.cs

```diff
@@ -0,0 +1,139 @@
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using System;
+using Coypu;
+using Remotion.ObjectBinding.Web.Contracts.DiagnosticMetadata;
+using Remotion.Utilities;
+using Remotion.Web.Development.WebTesting;
+
+namespace Remotion.ObjectBinding.Web.Development.WebTesting.ControlObjects
+{
+  /// <summary>
+  /// Represents information about validation error shown in the BocList.
+  /// </summary>
+  public class BocListValidationError
+  {
+    public static BocListValidationError Parse (ElementScope elementScope)
+    {
+      var errorMessage = elementScope.FindCss("span, a").InnerHTML ?? string.Empty; // .InnerHTML is used instead of .Text as it would return an empty string instead
+      var rowItemID = elementScope.GetAttribute(DiagnosticMetadataAttributesForObjectBinding.BocListValidationFailureSourceRow);
+      var columnItemID = elementScope.GetAttribute(DiagnosticMetadataAttributesForObjectBinding.BocListValidationFailureSourceColumn);
+      var businessObjectUniqueIdentifier = elementScope.GetAttribute(DiagnosticMetadataAttributesForObjectBinding.ValidationFailureSourceBusinessObject);
+      var propertyIdentifier = elementScope.GetAttribute(DiagnosticMetadataAttributesForObjectBinding.ValidationFailureSourceProperty);
+
+      return new BocListValidationError(
+          errorMessage,
+          string.IsNullOrEmpty(rowItemID) ? null : rowItemID,
+          string.IsNullOrEmpty(columnItemID) ? null : columnItemID,
+          string.IsNullOrEmpty(businessObjectUniqueIdentifier) ? null : businessObjectUniqueIdentifier,
+          string.IsNullOrEmpty(propertyIdentifier) ? null : propertyIdentifier);
+    }
+
+    /// <summary>
+    /// Gets the errors message associated with the validation error.
+    /// </summary>
+    public string ErrorMessage { get; }
+
+    /// <summary>
+    /// Gets the ItemID of the row in which the validation error occured,
+    /// or <see langword="null"/> if the validation error is not assigned to a row.
+    /// </summary>
+    public string? RowItemID { get; }
+
+    /// <summary>
+    /// Gets the ItemID of the column in which the validation error occured,
+    /// or <see langword="null"/> if the validation error is not assigned to a column.
+    /// </summary>
+    public string? ColumnItemID { get; }
+
+    /// <summary>
+    /// Gets the unique identifier of the business object associated with the validation error,
+    /// or <see langword="null"/> if the validation error has no associated business object.
+    /// </summary>
+    public string? BusinessObjectUniqueIdentifier { get; }
```

Comment:
We don't have a direct precedence, given property naming, `ValidatedDomainObject` is still best even with the amiguity with Remotion.Data.DomainObject.